notifications/README.md

```diff
@@ -0,0 +1,58 @@
+# Notifications
+
+## Notification Object
+
+```
+{
+  id: string,
+  username: string,
+  message: string,
+  createOn: unix-timestamp,
+  url: string,
```

Comment:
What does this url do?

Reply:
I was thinking that every notification could have some url associated with it. Like if it's coming from a task, we could provide the link to the task which has been unblocked for the user now. I'm not really sure how would we do that 😅

Reply:
I think the `message` field is generic enough for us to reuse as markdown for showing whatever links we would like

Reply:
> 
> 
> I think the `message` field is generic enough for us to reuse as markdown for showing whatever links we would like

An example, how will our backend get the `url` of that task which is unblocked now?

Reply:
Let's say Task-B depends on Task-A to be completed:  Task-A <- Task-B.
This dependency is stored by the tasks data model.

When Task-A is completed, it triggers an event E, which instructs our notifier to fire Notification N.

N pulls the details it needs from Task-A and Task-B to create its message (which may contain the details of the URL). N doesn't need an explicit `url` field to be maintained and synced. This avoids duplication and upkeep.